apps/taxes/src/modules/avatax/ui/avatax-tax-code-matcher-table.tsx

```diff
@@ -0,0 +1,116 @@
+import { useDashboardNotification } from "@saleor/apps-shared";
+import React from "react";
+import { trpcClient } from "../../trpc/trpc-client";
+import { Table } from "../../ui/table";
+import { Select } from "../../ui/_select";
+import { Box, Text } from "@saleor/macaw-ui/next";
+import { AppCard } from "../../ui/app-card";
+
+const SelectTaxCode = ({ taxClassId }: { taxClassId: string }) => {
+  const [value, setValue] = React.useState("");
+  const { notifySuccess, notifyError } = useDashboardNotification();
+
+  const { data: avataxMatches, isLoading: isMatchesLoading } =
+    trpcClient.avataxMatches.getAll.useQuery();
+
+  React.useEffect(() => {
+    if (avataxMatches) {
+      const match = avataxMatches?.find((item) => item.data.saleorTaxClassId === taxClassId);
+
+      if (match) {
+        setValue(match.data.avataxTaxCode);
+      }
+    }
+  }, [avataxMatches, taxClassId]);
```

Comment:
```suggestion
  const { data: avataxMatches, isLoading: isMatchesLoading } =
    trpcClient.avataxMatches.getAll.useQuery({
    onSuccess: (data) => {
      if(!data){
        return;
      }
      const match = data.find((item) => item.data.saleorTaxClassId === taxClassId);

      if (match) {
        setValue(match.data.avataxTaxCode);
      }
    }
    });
```

Not _wrong_, personally I find using onSuccess/error provided by react query more ergonomic to use. I see you used it in mutation